### PR TITLE
Fix #7214: install knows which ml files do not get compiled to cmx.

### LINKED ISSUE #7214: install complains because of a few not compiled ml files
eg in https://gitlab.com/SkySkimmer/coq/-/jobs/62215853
```
install: cannot stat 'clib/deque.cmi': No such file or directory
...
install: cannot stat 'clib/deque.cmx': No such file or directory
install: cannot stat 'plugins/extraction/big.cmx': No such file or directory
install: cannot stat 'toplevel/coqtop_byte_bin.cmx': No such file or directory
```
deque has never been used, not sure about the others.

### PR DESCRIPTION
**Pull request by SkySkimmer**
(we just delete the deque files since they're useless)

### CODE REVIEW

**Comment by advorkin** [comment]
Hello,

I removed deque.* and modified Makefile.install per your instructions.  Unfortunately, same problem.

Getting 910 cannot stat messages.  Ex. 
install: cannot stat 'theories/Reals/.coq-native/NCoq_Reals_DiscrR.o': No such file or directory

Using options -bytecode-compiler no -native-compiler yes

Thank you!

**Comment by SkySkimmer** [comment]
Then you want bug  #6354 ^^

**Comment by advorkin** [comment]
Thank you for responding.  But that particular bug uses  -native-compiler=no option, thus I wasn't sure if it's the same problem.

**Review by ppedrot — APPROVED** [review]
I don't remember why I introduced Deque in the first place, it might have been an incomplete patch that got in somehow. Anyways, it's fine to remove it as it's not used.

**Comment by SkySkimmer** [comment]
Can someone take charge of this PR? @ppedrot ?

**Comment by ppedrot** [comment]
I do not really understand the Makefile part, but I can take it if needed.

**Comment by ppedrot** [comment]
Maybe @gares or @ejgallego should have a look at this PR? They're more fluent in the build system than I am.

**Comment by gares** [comment]
I don't get why the Makefile should be touched by this PR.

**Comment by ppedrot** [comment]
@SkySkimmer Can you explain the Makefile part of the PR?

File: Makefile.install [hunk 1]
```diff
@@ -103,7 +103,9 @@ INSTALLCMI = $(sort \
 	$(foreach lib,$(CORECMA), $(addsuffix .cmi,$($(lib:.cma=_MLLIB_DEPENDENCIES))))) \
 	$(PLUGINS:.cmo=.cmi)
 
-INSTALLCMX = $(sort $(filter-out checker/% ide/% tools/% dev/% configure.cmx, $(MLFILES:.ml=.cmx)))
+INSTALLCMX = $(sort $(filter-out checker/% ide/% tools/% dev/% \
+	configure.cmx toplevel/coqtop_byte_bin.cmx plugins/extraction/big.cmx, \
+	$(MLFILES:.ml=.cmx)))
```
Comment (SkySkimmer) [inline Makefile.install]: MLFILES is gotten by `find . -name '*.ml'` and these specific files don't get compiled so we shouldn't try to install their cmx.

**Comment by Zimmi48** [comment]
@gares Does this explanation satisfy you?

File: Makefile.install (hunk 1 above)
Comment (gares) [reply Makefile.install]: My question is: why removing the `.ml` files this PR is removing is not enough?

**Comment by SkySkimmer** [comment]
Because coqtop_byte_bin is not removed

Gaëtan Gilbert

**Comment by gares** [comment]
ok

**Pull request merged**